launcher/mainwindow_moc.cpp

```diff
@@ -264,11 +269,25 @@ void MainWindow::dropEvent(QDropEvent* event)
 
 void MainWindow::manualInstallFile(QString filePath)
 {
-	if(filePath.endsWith(".zip", Qt::CaseInsensitive) || filePath.endsWith(".exe", Qt::CaseInsensitive))
+#ifdef VCMI_ANDROID
+	QString realFilePath{};
+	if(filePath.contains("content://", Qt::CaseInsensitive))
+	{
+		auto path = QAndroidJniObject::fromString(filePath);
+		realFilePath = QAndroidJniObject::callStaticObjectMethod("eu/vcmi/vcmi/util/FileUtil", "getFilenameFromUri", "(Ljava/lang/String;)Ljava/lang/String;", path.object<jstring>()).toString();
+	}
+	else
+		realFilePath = filePath;
+
+#else
+	QString realFilePath = filePath;
+#endif
```

Comment:
Extract to shared method? Copy-pasted at least once

Reply:
Yes, prefered file for placing this?

Reply:
No string preference. 
There is apparently `helper.h/cpp` pair in Launcher. Probably in there?